homeassistant/components/keyboard_remote.py

```diff
@@ -106,34 +82,60 @@ def _stop_keyboard_remote(_event):
 class KeyboardRemote(threading.Thread):
     """This interfaces with the inputdevice using evdev."""
 
-    def __init__(self, hass, device_descriptor, key_value):
+    def __init__(self, hass, config):
         """Construct a KeyboardRemote interface object."""
-        from evdev import InputDevice
+        from evdev import InputDevice, list_devices
 
-        self.device_descriptor = device_descriptor
-        try:
-            self.dev = InputDevice(device_descriptor)
-        except OSError:  # Keyboard not present
+        self.device_descriptor = config.get(DEVICE_DESCRIPTOR)
+        self.device_name = config.get(DEVICE_NAME)
+        self.dev = self._get_keyboard_device()
+        if self.dev is not None:
             _LOGGER.debug(
-                'KeyboardRemote: keyboard not connected, %s',
-                self.device_descriptor)
-            self.keyboard_connected = False
+                'KeyboardRemote: keyboard connected'
```

Comment:
Also, to clean up the existing code, let's remove the prefix "KeyboardRemote: " from all of these log messages. Logs are already identified by their platform.